library-best-practices/JLBP-14.md

```diff
@@ -0,0 +1,16 @@
+[JLBP-14] Do not use version ranges
+-----------------------------------
+
+- If the pom.xml of your library specifies a version range instead of a
+  single version for any particular dependency, builds at different points
+  in time can get different versions of that dependency (as your dependency
+  releases new versions), which could break your library unexpectedly.
+  - Version ranges can even pull in incomplete releases that break your
+    library's build (example:
```

Comment:
Random thought: did this qualify as a static linkage error or it a different kind of problem?

Reply:
I would say this is a different type of problem; the dependency tree that was specified couldn't be fetched.